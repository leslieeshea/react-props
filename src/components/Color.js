import React from 'react';
import PropTypes from 'prop-types';

function Color({ name, hex, red, green, blue }) {
  return (
    <dl>
      <dt>Name</dt>
      <dd>{name ? name : hex}</dd>

      <dt>Hex</dt>
      <dd>{hex}</dd>

      <dt>RGB</dt>
      <dd>Red - {red}</dd>
      <dd>Green - {green}</dd>
      <dd>Blue - {blue}</dd>
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string,
  hex: PropTypes.string.isRequired,
  red: PropTypes.number.isRequired,
  green: PropTypes.number.isRequired,
  blue: PropTypes.number.isRequired
};

export default Color;
