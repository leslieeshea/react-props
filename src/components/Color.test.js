import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color Component', () => {
  it('renders a color', () => {
    const wrapper = shallow(<Color name="Red" rgb={{ red: 255, green: 0, blue: 0 }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
