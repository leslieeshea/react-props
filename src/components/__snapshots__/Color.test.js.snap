// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Component renders a color 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    Red
    <div
      style={
        Object {
          "backgroundColor": "#FF0000",
          "display": "inline-block",
          "height": "0.8em",
          "width": "0.8em",
        }
      }
    />
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #FF0000
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    Red - 
    255
  </dd>
  <dd>
    Green - 
    0
  </dd>
  <dd>
    Blue - 
    0
  </dd>
</dl>
`;
